pkg/proxy/nftables/proxier.go

```diff
@@ -1094,6 +1203,14 @@ func (proxier *Proxier) syncProxyRules() {
 			proxier.logger.Error(nil, "Failed to cast serviceInfo", "serviceName", svcName)
 			continue
 		}
+
+		// svcTx is used for all service-related chains and their elements.
+		// If no changes were done to the service or its endpoints, these objects may be skipped.
+		svcTx := tx
+		if tryPartialSync && !serviceUpdateResult.UpdatedServices.Has(svcName.NamespacedName) && !endpointUpdateResult.UpdatedServices.Has(svcName.NamespacedName) {
+			svcTx = nil
+		}
```

Comment:
So in iptables kube-proxy there are two weird constraints on partial syncs:
1. Even when doing a partial sync, we have to write out the entire `KUBE-SERVICES`, `KUBE-NODEPORTS`, etc chains, including the rules for services that didn't change. (This happens in the part of the loop before the "tryPartialSync and !UpdatedServices.Has()" check.)
2. Even when doing a partial sync, we have to compute the `sync_proxy_rules_iptables_total` metric, meaning we have to "pretend" to generate all of the rules even though we don't actually write them all to iptables. (This happens in the part of the loop _after_ the "tryPartialSync and !UpdatedServices.Has()" check.)

nftables kube-proxy should not have either of those problems. I *think* that if this condition is true, you can just `continue` from here (or a few lines earlier) and skip the entire loop for this servicePort. (That's definitely what we're aiming for eventually anyway.)

Reply:
We still need to update `clusterIPs`, ..., `serviceNodePorts` maps to ensure we don't delete related elements in the end of the sync. Same for `activeAffinitySets` and `activeChains`.
I can try to reorder the operations in this loop, to return after all non-svcTx operations are done, wdyt?

Reply:
yes; that was how it originally worked in iptables, before we realized we needed to compute the metric too. See #110266. The nftables loop was just copied from the iptables code, so it's currently in "the order that made sense for iptables", but we might need to reorder it a little for nftables.